docs/upgrade/upgrade-7.17-8.x.asciidoc

```diff
@@ -0,0 +1,155 @@
+
+. Run the {kib-ref}/upgrade-assistant.html[Upgrade Assistant]. Take note of any critical error messages, then work with your Support rep to resolve them.
++
+.Requirements
+[sidebar]
+--
+To run the Upgrade Assistant, you must have `superuser` cluster admin permissions.
+--
+
+. If you're not using {ref}/snapshots-take-snapshot.html#automate-snapshots-slm[{slm} (SLM)], you must set up, configure, and run at least one {ref}/snapshot-restore.html[snapshot]. Snapshots are {ref}/snapshot-restore.html#how-snapshots-work[incremental] – depending on the cluster size and the input/output rate, the initial snapshot may take several hours to complete. If you are using SLM, {ref}/snapshots-take-snapshot.html#check-slm-history[check the SLM history] to ensure that snapshots are completing successfully.
+
+[float]
+=== Perform 8.x upgrade on Non-Prod (Optional) or Prod Deployment
+
+NOTE: We strongly recommend you perform the following steps on a non-prod test deployment first to address any potential issues before upgrading on a prod deployment. 
+
+. Back up all of your data to a {ref}/snapshots-take-snapshot.html[snapshot] – which is a backup of an index taken from a running cluster. Rollbacks require a snapshot and cause data loss without one.
+
+. We recommend you <<rules-api-export, export all your custom detection rules>> in the event there are issues with the detection engine after the upgrade.
+
+. Upgrade {es}. Refer to {stack-ref}/upgrading-elasticsearch.html[these upgrade instructions]. 
+** If you're using {ecloud}, we recommend upgrades with no downtime. Refer to {cloud}/ec-upgrade-deployment.html[these instructions].  
+** If you're using {ece} (ECE), refer to {ece-ref}/ece-upgrade-deployment.html[these instructions].  
+** If you're using {eck} (ECK), refer to {eck-ref}/k8s-upgrading-stack.html[these instructions]. 
+** If you're upgrading a self-orchestrated deployment, upgrade the data nodes first, tier-by-tier, in this order:
+... Frozen tier
+... Cold tier 
+... Warm tier
+... Hot tier 
+... Any other nodes not in a tier
+... All remaining nodes that are neither master-eligible nor data nodes
+... Master-eligible nodes
+
+. Upgrade {kib}. Refer to {stack-ref}/upgrading-kibana.html[these instructions].
++
+NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.
+
+. Validate {es} and {kib} are operating as expected by completing the following checks: 
+.. {es}
+... Check the status of your clusters and ensure they're green. Run the `GET _cat/health` API request. For more information, refer to the {ref}/cat-health.html[cat health API documentation].
+... Check the size of your rolled over indices/shards, which should be less than 50 GB. Run the `GET _cat/shards?s=store:desc` API request. For more information, refer to the {ref}/cat-shards.html[cat shards API documentation].    
```

Comment:
Not clear about the 50GB mention here. Is this something users need to ensure at this time in the process? What if they have shards that are larger than 50GB? (Google tells me there's no hard limit on shard size.)

Reply:
Going to remove this statement about checking the size, per @shainaraskas' recommendation.